packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1243 @@
+                break;
+            case TableFocusType.cell: {
+                focusableElement = rowElements.cells[this.columnIndex]!.cell;
+                break;
+            }
+            case TableFocusType.cellActionMenu:
+                focusableElement = rowElements.cells[this.columnIndex]?.actionMenuButton;
+                break;
+            case TableFocusType.cellContent: {
+                focusableElement = rowElements.cells[this.columnIndex]?.cell.cellView
+                    .tabbableChildren[this.cellContentIndex];
+                break;
+            }
+            default:
+                break;
+        }
+        if (focusableElement) {
+            this.focusElement(focusableElement, focusOptions);
+        }
+    }
+
+    private focusHeaderElement(): boolean {
+        const headerElements = this.getTableHeaderFocusableElements();
+        let focusableElement: HTMLElement | undefined;
+        switch (this.focusType) {
+            case TableFocusType.headerActions:
+                focusableElement = headerElements.headerActions[this.headerActionIndex];
+                break;
+            case TableFocusType.columnHeader:
+                focusableElement = headerElements.columnHeaders[this.columnIndex];
+                break;
+            default:
+                break;
+        }
+        if (focusableElement) {
+            this.focusElement(focusableElement);
+            return true;
+        }
+        return false;
+    }
+
+    private getCurrentRowVisibleIndex(): number {
+        return this.table.rowElements.findIndex(
+            row => row.rowStateIndex === this.rowIndex
+        );
+    }
+
+    private getTableHeaderFocusableElements(): TableHeaderFocusableElements {
+        const headerActions: HTMLElement[] = [];
+        if (
+            this.table.selectionCheckbox?.getRootNode()
+            === this.table.shadowRoot
+        ) {
+            headerActions.push(this.table.selectionCheckbox);
+        }
+
+        if (
+            this.table.showCollapseAll
+            && this.table.collapseAllButton?.getRootNode()
+                === this.table.shadowRoot
+        ) {
+            headerActions.push(this.table.collapseAllButton);
+        }
+
+        const columnHeaders: HTMLElement[] = [];
+        if (this.canFocusColumnHeaders()) {
+            this.table.columnHeadersContainer
+                .querySelectorAll(tableHeaderTag)
+                .forEach(header => columnHeaders.push(header));
+        }
+
+        return { headerActions, columnHeaders };
+    }
+
+    private canFocusColumnHeaders(): boolean {
+        return this.table.columns.find(c => !c.sortingDisabled) !== undefined;
+    }
+
+    private getCurrentRow(): TableRow | TableGroupRow | undefined {
+        const visibleRowIndex = this.getCurrentRowVisibleIndex();
+        if (visibleRowIndex >= 0) {
+            return this.table.rowElements[visibleRowIndex];
+        }
+        return undefined;
```

Comment:
Don't feel strongly, but I think you can just ditch the `if` block.

Reply:
Done